google/cloud/bigtable/client.py

```diff
@@ -94,8 +310,40 @@ def __init__(
         instance_id: str,
         table_id: str,
         app_profile_id: str | None = None,
+        metadata: list[tuple[str, str]] | None = None,
     ):
-        raise NotImplementedError
+        """
+        Initialize a Table instance
+
+        Must be created within an async run loop context
+
+        Args:
+            instance_id: The Bigtable instance ID to associate with this client
+                instance_id is combined with the client's project to fully
+                specify the instance
+            table_id: The ID of the table.
+            app_profile_id: (Optional) The app profile to associate with requests.
+                https://cloud.google.com/bigtable/docs/app-profiles
+            metadata: a list of metadata headers to be attached to all calls with this client
+        Raises:
+          - RuntimeError if called outside of an async run loop context
+        """
+        self.client = client
+        self.instance = instance_id
+        self.table_id = table_id
+        self.app_profile_id = app_profile_id
+        self.metadata = metadata or []
+        # raises RuntimeError if called outside of an async run loop context
+        try:
+            self._register_instance_task = asyncio.create_task(
+                self.client.register_instance(instance_id)
+            )
+        except RuntimeError:
+            warnings.warn(
+                "Table should be created in an asyncio event loop."
+                " Instance will not be registered with client for refresh",
+                RuntimeWarning,
+            )
```

Comment:
why not bubble up this error?

Reply:
I've gone back and forth on it

The question comes down to whether we should support creating these classes outside async context:
- if there's not an active event loop, we can't start our background channel management coroutines at init time, but everything else should work
- I don't think it's a best practice to enforce only initializing objects with an active event loop, so I opted to raise a warning instead. Users can always manually start the coroutines in an async context later
- That said, I'm not sure what the use-cases are for initializing outside of an async context, so I'm not sure how important this is to support
- I've considered using a background thread for channel management, which should avoid this problem, but I was hoping we could get by with just coroutines to avoid the complication of multiple concurrency methods

Let me know if you have thoughts